src/jaxsim/simulation/integrators.py

```diff
@@ -22,27 +23,38 @@
     [State, Time], Tuple[StateDerivative, Dict[str, Any]]
 ]
 
+Carry = Tuple[State, Time]
```

Comment:
This is an implementation detail, can you move it into the step function again?

Reply:
Done in https://github.com/ami-iit/jaxsim/pull/82/commits/360d611dee725fdcc4968142873ce3765057ff07